HomeScreen.js

```diff
@@ -54,6 +61,14 @@ function SearchIcon(props) {
 
 function FavoriteCities() {
   const favorites = useFavorites();
+
+  // Add the stats for each city
+  favorites.map(favorite => {
+    favorite.stats = useCityStats(favorite.cityId);
```

Comment:
This will cause errors because the useCityStats hook is being used inside a loop. See more details here: 
https://reactjs.org/docs/hooks-rules.html

Also, the useCityStats data never gets updated, so caching behavior should be implemented so the avgPM2_5 data does not get old

Reply:
Ok, I commited a refactored version. I don't understand the caching issue, can you elaborate?